aries_vcx/src/protocols/issuance/holder/state_machine.rs

```diff
@@ -288,9 +302,21 @@ impl HolderSM {
                     HolderFullState::RequestSent(state_data)
                 }
             },
-            HolderFullState::Finished(state_data) => {
-                warn!("Unable to process received message in this state");
-                HolderFullState::Finished(state_data)
+            HolderFullState::Finished(state_data) => match cim {
+                CredentialIssuanceAction::RevocationNotification(notification) => {
+                    if notification.ack_on(AckOn::Receipt) {
+                        let ack = build_revocation_ack(&notification.get_thread_id());
```

Comment:
if you agree with my previous comment that notification should have no `thread_id`, we would have to build thread_id here from `@id` of received message